src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsworkload/config/WebClientUserEnhancementConfiguration.kt

```diff
@@ -45,60 +44,62 @@ class WebClientUserEnhancementConfiguration(
+    val reactiveClientCredentialsTokenResponseClient = WebClientReactiveClientCredentialsTokenResponseClient()
 
-    val defaultClientCredentialsTokenResponseClient = DefaultClientCredentialsTokenResponseClient()
+    reactiveClientCredentialsTokenResponseClient.setWebClient(builder.filter(userEnhancedTokenRequestProcessor()).build())
 
-    val authentication = SecurityContextHolder.getContext().authentication
+    val reactiveAuthorizedClientProvider = ReactiveOAuth2AuthorizedClientProviderBuilder
+      .builder()
+      .clientCredentials { reactiveClientCredentialsGrantBuilder: ReactiveOAuth2AuthorizedClientProviderBuilder.ClientCredentialsGrantBuilder ->
+        reactiveClientCredentialsGrantBuilder.accessTokenResponseClient(reactiveClientCredentialsTokenResponseClient)
+      }.build()
 
-    defaultClientCredentialsTokenResponseClient.setRequestEntityConverter { grantRequest: OAuth2ClientCredentialsGrantRequest ->
-      val converter = CustomOAuth2ClientCredentialsGrantRequestEntityConverter()
-      val username = authentication.name
-      converter.enhanceWithUsername(grantRequest, username)
-    }
+    manager.setAuthorizedClientProvider(reactiveAuthorizedClientProvider)
+    return manager
+  }
 
-    val authorizedClientProvider = OAuth2AuthorizedClientProviderBuilder.builder()
-      .clientCredentials { clientCredentialsGrantBuilder: OAuth2AuthorizedClientProviderBuilder.ClientCredentialsGrantBuilder ->
-        clientCredentialsGrantBuilder.accessTokenResponseClient(defaultClientCredentialsTokenResponseClient)
+  fun userEnhancedTokenRequestProcessor(): ExchangeFilterFunction = ExchangeFilterFunction.ofRequestProcessor { request ->
+    ReactiveSecurityContextHolder.getContext()
+      .map(SecurityContext::getAuthentication)
+      .map(Authentication::getName)
+      .map { username ->
+        val clientRequest = ClientRequest.from(request)
+        if (request.body() is BodyInserters.FormInserter<*>) {
+          clientRequest.body((request.body() as BodyInserters.FormInserter<String>).with("username", username))
+        }
+        clientRequest.build()
       }
-      .build()
-
-    manager.setAuthorizedClientProvider(authorizedClientProvider)
-    return manager
   }
 
   private fun getOAuthWebClient(
-    authorizedClientManager: OAuth2AuthorizedClientManager,
+    authorizedClientManager: ReactiveOAuth2AuthorizedClientManager,
     builder: WebClient.Builder,
     rootUri: String,
     registrationId: String
   ): WebClient {
-    val oauth2Client = ServletOAuth2AuthorizedClientExchangeFilterFunction(authorizedClientManager)
+    val oauth2Client = ServerOAuth2AuthorizedClientExchangeFilterFunction(authorizedClientManager)
     oauth2Client.setDefaultClientRegistrationId(registrationId)
     return builder.baseUrl(rootUri)
-      .apply(oauth2Client.oauth2Configuration())
+      .filter(oauth2Client)
       .build()
   }
 
   @Bean
   fun assessRiskNeedsApiWebClient(builder: WebClient.Builder): WebClient {
```

Comment:
changed to use filter to pass through the Auth Token rather than getting it in the controller and passing it through the stack of calls